js_modules/dagster-ui/packages/ui-core/src/assets/__tests__/RecentlyVisitedAssetsStorage.test.tsx

```diff
@@ -0,0 +1,112 @@
+import {MockedProvider} from '@apollo/client/testing';
+import {render, screen, waitFor} from '@testing-library/react';
+import Router, {MemoryRouter} from 'react-router-dom';
+
+import {AnalyticsContext} from '../../app/analytics';
+import {AssetLiveDataProvider} from '../../asset-data/AssetLiveDataProvider';
+import {buildAsset, buildAssetKey} from '../../graphql/types';
+import {buildQueryMock} from '../../testing/mocking';
+import AssetsCatalogRoot, {ASSETS_CATALOG_ROOT_QUERY} from '../AssetsCatalogRoot';
+import {fetchRecentlyVisitedAssetsFromLocalStorage} from '../RecentlyVisitedAssetsStorage';
+import {
+  AssetsCatalogRootQuery,
+  AssetsCatalogRootQueryVariables,
+} from '../types/AssetsCatalogRoot.types';
+
+// Mock the `useParams` hook to override the asset key in the URL.
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: jest.fn(),
+}));
+
+describe('RecentlyVisitedAssetsStorage', () => {
+  beforeEach(() => {
+    window.localStorage.clear();
+  });
+
+  it('Store visited asset in local storage', async () => {
+    // Set the asset key in the URL
+    jest.spyOn(Router, 'useParams').mockReturnValue({0: 'sda_asset'});
+
+    const assetKey = {path: ['sda_asset']};
+    const TestAssetView = (
+      <MockedProvider
+        mocks={[
+          buildQueryMock<AssetsCatalogRootQuery, AssetsCatalogRootQueryVariables>({
+            query: ASSETS_CATALOG_ROOT_QUERY,
+            variables: {
+              assetKey,
+            },
+            data: {
+              assetOrError: buildAsset({
+                key: buildAssetKey(assetKey),
+              }),
+            },
+          }),
+        ]}
+      >
+        <AnalyticsContext.Provider value={{page: () => {}} as any}>
+          <AssetLiveDataProvider>
+            <MemoryRouter initialEntries={['/assets/sda_asset']}>
+              <AssetsCatalogRoot />
+            </MemoryRouter>
+          </AssetLiveDataProvider>
+        </AnalyticsContext.Provider>
+      </MockedProvider>
+    );
+
+    // First render displays the page in a loading state
+    const {rerender} = render(TestAssetView);
+    await waitFor(() => {
+      expect(screen.queryByText('Loading assets…')).toBeNull();
+    });
+    // Second render displays the page in a loaded state
```

Comment:
`expect(fetchRecentlyVisitedAssetsFromLocalStorage()).toEqual([]);` to confirm it doesn't save it until after the loading finishes